src/miniscript/context.rs

```diff
@@ -753,7 +800,13 @@ impl ScriptContext for BareCtx {
         match &ms.node {
             Terminal::Check(ref ms) => match &ms.node {
                 Terminal::RawPkH(_pkh) => Ok(()),
-                Terminal::PkK(_pk) | Terminal::PkH(_pk) => Ok(()),
+                Terminal::PkH(_pk) => Ok(()),
+                Terminal::PkK(key) => match key {
+                    KeyExpr::<Pk>::SingleKey(_pk) => Ok(()),
+                    KeyExpr::<Pk>::MuSig(_) => {
+                        return Err(Error::NonStandardBareScript);
```

Comment:
You are returning an incorrect variant here

Reply:
changed it to
`Err(Error::ContextError(ScriptContextError::MusigNotAllowed(String::from(Self::name_str()))))`